### PR TITLE
Add papertrail admin interface

### PR DESCRIPTION
**Pull request by Jeremy1026**
### Relevant Ticket or Conversation:

[LG-12358](https://cm-jira.usa.gov/browse/LG-12358)

### Description of Changes:

Adds a section for version history on the integration configuration page only visible to admins. Shows changes made, grouped by version.

image

### PR Checklist:

1. [ ] Have you linted and tested your code locally prior to submission?
2. [ ] Have you tagged the appropriate dev(s) for review?
3. [ ] Have you linked to any relevant tickets or conversations?

### PR Review Standards: 
- Consider using [Conventional Comments](https://conventionalcomments.org/) to ensure that your feedback is clear and actionable.
- Ideally, PRs should be reviewed by at least 2 team members.
- All PRs must be approved before being merged.

### CODE REVIEW

File: app/views/service_providers/show.html.erb [hunk 1]
```diff
@@ -222,3 +222,44 @@
...
+            <thead>
+              <tr>
+                <th scope="col">Field</th>
+                <th scope="col">Previous Value</th>
+                <th scope="col">Updated Value</th>
+              </tr>
+            </thead>
+          <% v.object_changes.each do |field, change| %>
+            <% if field != 'updated_at' %>
+              <% before, after = change %>
```
Comment (zachmargolis) [inline app/views/service_providers/show.html.erb:249]: we can destructure inside the loop
```suggestion
          <% v.object_changes.each do |field, (before, after)| %>
            <% if field != 'updated_at' %>
```

File: app/views/service_providers/show.html.erb [hunk 2]
```diff
@@ -222,3 +222,44 @@
...
+    <button class="usa-accordion__button margin-top-105" aria-expanded="false" aria-controls="versions">
+      <h3 id="versionToggle" class="usa-accordion__heading">Version History (<%= service_provider.versions.count %>) </h3>
+    </button>
+    <div id="versions" class="usa-accordion__content">
+    <% service_provider.versions.reverse.each do |v| %>
+      <div class="event">
+        <h4 class="margin-0">Action: <%= v.event.capitalize %></h4>
+        <h5 class="margin-0">By: <%= v.whodunnit %></h5>
+        <h5 class="margin-0">At: <%= v.created_at %></h5>
+        <% if v.event != 'create' %>
```
Comment (Sgtpluck) [inline app/views/service_providers/show.html.erb:238]: [question] is there a reason to leave out the initial create? probably helpful to know the starting point!

Comment (Jeremy1026) [reply app/views/service_providers/show.html.erb:238]: My thought process there was:

* The create list is rather long
* We can go through changes to get back to the initial values
* If the list of versions is super long making reversing the changes to get to the original difficult, knowing what the original value was probably isn't useful

File: app/views/service_providers/show.html.erb (hunk 1 above)
Comment (Jeremy1026) [reply app/views/service_providers/show.html.erb]: I like simpler! Committed.

File: app/views/service_providers/show.html.erb (hunk 2 above)
Comment (Sgtpluck) [reply app/views/service_providers/show.html.erb:238]: that makes sense. i think there is possibly some utility in knowing the first iteration, but that's just a guess, so we can add it if we need it! thanks 🙏

**Review by Sgtpluck — APPROVED** [review]

**Pull request merged**